### ISSUE TITLE
Quick start: I think make -j is not a good idea

### ISSUE THREAD

**Issue by maxballenger** [grpc/grpc.io#740]
I don't think "make -j" should be recommended in the quick start guide. I just ran this on a moderately resourced VM (4 cores, 8 GB memory) and it ran the VM out of memory, leaving it inaccessible by SSH, probably because the SSH server got starved. I logged in through VMware Workstation and was able to see that there were dozens of cc1plus processes running. "make -j nproc" would be safer I think.

**Comment by sanjaypujare** [comment]
cc @yashykt

**Comment by yashykt** [comment]
I'll note that the instructions in the quick start guide would not work for everyone. In some cases, we do need to tweak the instructions. I am fine with removing `- j` from the instructions in this case. Please feel free to send a PR for this.

**Comment by chalin** [comment]
Thanks all for your comments.

I'd rather not drop the `-j` option altogether: doing so makes the build painfully slow IMHO -- especially for a _quick_ start. I prefer @maxballenger's suggestion of including a bound: `-j some-number`. Or we could add a [`-l/--max-load` option][max-load].

@jtattermusch - any thoughts?

[max-load]: https://www.gnu.org/software/make/manual/html_node/Parallel.html#Parallel

**Comment by jtattermusch** [comment]
> Thanks all for your comments.
> 
> I'd rather not drop the `-j` option altogether: doing so makes the build painfully slow IMHO -- especially for a _quick_ start. I prefer @maxballenger's suggestion of including a bound: `-j some-number`. Or we could add a [`-l/--max-load` option](https://www.gnu.org/software/make/manual/html_node/Parallel.html#Parallel).
> 
> @jtattermusch - any thoughts?

Yeah, I'd keep `-j` in some form (`-j4` seems fair?). Leaving the resolution up to you. Btw, seeing the machine go OOM because of `-j` seems odd - how many paraller worker did make end up using?

**Comment by maxballenger** [comment]
I don't know exactly how many parallel workers make ended up using but it
was basically filling up the whole terminal window when I ran top. I think
I saw at least 20.
